src/ParamConverter/DeveloperAppNameConverter.php

```diff
@@ -87,7 +88,12 @@ public function convert($value, $definition, $name, array $defaults) {
           // Load the entity directly from Apigee Edge if needed.
           // @see \Drupal\apigee_edge\ParamConverter\ApigeeEdgeLoadUnchangedEntity
           if (!empty($defaults['_route_object']->getOption('apigee_edge_load_unchanged_entity'))) {
-            $entity = $app_storage->loadUnchanged($app_id);
+            if ($app_storage instanceof AppStorage) {
+              $entity = $app_storage->loadUnchangedById($app_id);
+            }
+            else {
+              $entity = $app_storage->loadUnchanged($app_id);
```

Comment:
I think this bit could use some work. What is `$app_storage` going to be in the alternate? `$app_storage->loadUnchanged`is going to fail if `$app_storage` is not an instance of `AppStorage` right?

Reply:
It is not going to fail, that alternative is the current only alternative. It should be replaced with loadUnchangedByUuid to make it more efficient, but this new method is available only on AppStorage. There is not hard guarantee that $app_storage will always be an AppStorage, thus the check.

See this note by @mxr576 : "We need a TODO here because this method should be also available in the AppStorageInterface but that would be a breaking change so we can only add that in the next major version of the module."  https://github.com/apigee/apigee-edge-drupal/pull/460#discussion_r482836296

Reply:
I see. Thank you.

Reply:
A `if (method_exists($app_storage, 'loadUnchangedById'))` might work as well.

Reply:
Could you be more specific, yes it might work, but current also works. Does your comment has a particular purpose? Why is that better?